CMakeLists.txt

```diff
@@ -0,0 +1,118 @@
+cmake_minimum_required(VERSION 3.5)
+project(Surge VERSION 1.0.0 LANGUAGES CXX)
+
+set(SURGE_COMMON_SOURCES
+    src/common/dsp/effect/ConditionerEffect.cpp
+    src/common/dsp/effect/DistortionEffect.cpp
+    src/common/dsp/effect/DualDelayEffect.cpp
+    src/common/dsp/effect/Effect.cpp
+    src/common/dsp/effect/FreqshiftEffect.cpp
+    src/common/dsp/effect/PhaserEffect.cpp
+    src/common/dsp/effect/Reverb1Effect.cpp
+    src/common/dsp/effect/Reverb2Effect.cpp
+    src/common/dsp/effect/RotarySpeakerEffect.cpp
+    src/common/dsp/effect/VocoderEffect.cpp
+    src/common/dsp/AdsrEnvelope.cpp
+    src/common/dsp/BiquadFilter.cpp
+    src/common/dsp/BiquadFilterSSE2.cpp
+    src/common/dsp/DspUtilities.cpp
+    src/common/dsp/FilterCoefficientMaker.cpp
+    src/common/dsp/FMOscillator.cpp
+    src/common/dsp/LfoModulationSource.cpp
+    src/common/dsp/Oscillator.cpp
+    src/common/dsp/QuadFilterChain.cpp
+    src/common/dsp/QuadFilterUnit.cpp
+    src/common/dsp/SampleAndHoldOscillator.cpp
+    src/common/dsp/SurgeSuperOscillator.cpp
+    src/common/dsp/SurgeVoice.cpp
+    src/common/dsp/VectorizedSvfFilter.cpp
+    src/common/dsp/Wavetable.cpp
+    src/common/dsp/WavetableOscillator.cpp
+    src/common/dsp/WindowOscillator.cpp
+    src/common/thread/CriticalSection.cpp
+    src/common/util/FpuState.cpp
+    src/common/vt_dsp/basic_dsp.cpp
+    src/common/vt_dsp/halfratefilter.cpp
+    src/common/vt_dsp/lipol.cpp
+    src/common/vt_dsp/macspecific.cpp
+    src/common/Parameter.cpp
+    src/common/precompiled.cpp
+    src/common/Sample.cpp
+    src/common/SampleLoadRiffWave.cpp
+    src/common/SurgeError.cpp
+    src/common/SurgePatch.cpp
+    src/common/SurgeStorage.cpp
+    src/common/SurgeStorageLoadWavetable.cpp
+    src/common/SurgeSynthesizer.cpp
+    src/common/SurgeSynthesizerIO.cpp
+    src/common/UserDefaults.cpp
+    libs/xml/tinyxml.cpp
+    libs/xml/tinyxmlerror.cpp
+    libs/xml/tinyxmlparser.cpp
+    libs/filesystem/filesystem.cpp
+)
+
+set(SURGE_COMMON_INCLUDES
+    src/common
+    src/common/dsp
+    src/common/gui
+    libs/xml
+    src/common/vt_dsp
+    src/common/thread
+    vst3sdk/vstgui4
```

Comment:
This should definitely be vstgui.surge not vst3sdk/vstgui4 so we get our overridden vstgui